model/task_lifecycle.go

```diff
@@ -628,32 +630,17 @@ func doBisectStepback(ctx context.Context, t *task.Task) error {
 	if nextTask == nil {
 		return errors.Errorf("midway task could not be found for tasks '%s' '%s'", s.LastFailingStepbackTaskId, s.LastPassingStepbackTaskId)
 	}
-	// Log our next task to our last passing and last failing.
-	for _, id := range []string{s.LastFailingStepbackTaskId, s.LastPassingStepbackTaskId} {
-		if t, err := task.FindOneId(id); err != nil {
-			if t.StepbackInfo == nil {
-				grip.Warning(message.Fields{
-					"message":         "task in impossible state while performing bisection stepback",
-					"on_last_passing": id == s.LastPassingStepbackTaskId,
-					"last_passing_id": s.LastPassingStepbackTaskId,
-					"on_last_failing": id == s.LastFailingStepbackTaskId,
-					"last_failing_id": s.LastFailingStepbackTaskId,
-					"current_task":    nextTask.Id,
-				})
-				continue
-			}
-			// If the next task is already set, then this task has been used for stepback before
-			// and should not be updated (e.g. the right bound will have the same next stepback taskid
-			// as the left bound narrows the scope in which stepback happens).
-			if t.StepbackInfo.NextStepbackTaskId != "" {
-				continue
-			}
-			info := *t.StepbackInfo
-			info.NextStepbackTaskId = nextTask.Id
-			if err = task.SetStepbackInfo(s.LastFailingStepbackTaskId, info); err != nil {
-				return errors.Wrapf(err, "setting stepback info for task '%s'", nextTask.Id)
-			}
-		}
+	s.NextStepbackTaskId = nextTask.Id
+	// Store our next task to our last passing and last failing.
+	if err := task.SetNextStepbackId(s.LastFailingStepbackTaskId, s); err != nil {
```

Comment:
We don't need to set the next stepback task Id field on both `LastFailingStepbackTaskId` and `LastPassingStepbackTaskId`, just the current task `t`, as the field should just be pointing to the next task being activated in the stepback process

Reply:
We just need `NextStepbackTaskId ` and `PrevStepbackTaskId ` to construct a linkage of activation between the stepback tasks (i.e. 0<->4<->2<->3 for the first example in this [doc](https://docs.google.com/document/d/17bMlUDupaom5p9uIL1UbECD_XNf7POWhDNf0efv_VZk/edit))

Reply:
I had assumed that for the case of the first iteration, we want to link the last passing task as well to the stepback- I removed it now, so it can only be tracked from that head task that initiated stepback